### ISSUE TITLE
s3-data-plane documentation

### ISSUE THREAD

**Issue by reisman234** [eclipse-dataspaceconnector/DataSpaceConnector#1193]
Hello, 

I want to try out the data-plane-s3 extension, but the readme file is very short and does not contain any information about how to use it. 
At the moment, I just don't know how to specify the s3 endpoint and a bucket name. I think the credentials for the bucket can be provided as an environment, but please correct me if I'm wrong.

ralph

**Comment by ndr-brt** [comment]
The `data-plane-s3` extension was implemented mainly to move the communication logic away from the deprecated `InlineDataFlowController` to the `data-plane`, it's not documented nor tested appropriately.

[Currently](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/extensions/aws/data-plane-s3/src/main/java/org/eclipse/dataspaceconnector/aws/dataplane/s3/DataPlaneS3Extension.java#L44
), the credentials are obtained through the aws-sdk's [`DefaultCredentialsProvider`](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/DefaultCredentialsProvider.html), that loads the credentials in this order (according to the documentation):

```
AWS credentials provider chain that looks for credentials in this order:
- Java System Properties - aws.accessKeyId and aws.secretAccessKey
- Environment Variables - AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
- Credential profiles file at the default location (~/.aws/credentials) shared by all AWS SDKs and the AWS CLI
- Credentials delivered through the Amazon EC2 container service if AWS_CONTAINER_CREDENTIALS_RELATIVE_URI" environment variable is set and security manager has permission to access the variable,
- Instance profile credentials delivered through the Amazon EC2 metadata service
```

If you are interested in contributing with testing and documenting would be great.

**Comment by reisman234** [comment]
Thanks for the response! I did a first small test with the `data-plane-s3` extension, in which I copy a file from s3 to local. You can find  it in my fork and `s3-test` branch.  https://github.com/reisman234/DataSpaceConnector/tree/s3-test

I think also some parts in the s3-core needs some updates, e.g. the `S3ClientProviderImpl.java` does not provide functionality to choose a self-hosted S3-Server (minio). 

I will see what I can do for testing and documentation. What resources do you expect as a test for this extension? Just Unit tests, or is such an example also sufficient? 

In general to that data-plan topic, I have another question. In the file transfer samples, only the provider realizes the data plane part. How can a consumer download a file after the negotiation part?

**Comment by ndr-brt** [comment]
I created an issue about the provider being able to create a client to connect to a self hosted S3 (#1309).

I think the best test would be an end to end one, that uses a dokerized minio to verify the data transfer (as already done by the S3 integration-tests).

**Comment by reisman234** [comment]
I created a patch in my fork for this issue https://github.com/reisman234/DataSpaceConnector/tree/feature/%231309-configurable-s3-endpoint, if you want to have a look. 

In a first manual test, I was able to do a provider push from s3 to a http endpoint.

What can I use for as a reference for doing end-to-end testing? I found some resources in `./system-tests/e2e-transfer-test/`  but I don't know how to tun them.
